starter-code/database.js

```diff
@@ -133,27 +134,33 @@ class Database {
         // productName is the name of the product that we want to buy
         // Think if you may need to implement two queries chained
         // remeber once it's finish to comment callback('Error buying product');
-        
-        callback('Error buying product');
+        database.collection('products').find({"name" : productName}).toArray((error, product) => {                    database.collection('users').update({'firstName' : userFirstName}, {$push : {"shoppingCart" : product[0]._id}},null, callback);
```

Comment:
In this case you may have just used the findOneAndUpdate method, shorter way to do the same